Source/JavaScriptCore/dfg/DFGStrengthReductionPhase.cpp

```diff
@@ -916,7 +916,7 @@ class StrengthReductionPhase : public Phase {
                 }
                 m_graph.watchpoints().addLazily(globalObject->regExpRecompiledWatchpoint());
                 regExp = regExpObjectNode->castOperand<RegExp*>();
-            } else if (String searchString = m_node->child2()->tryGetString(m_graph); !!searchString) {
+            } else if (String searchString = m_node->child2()->tryGetString(m_graph); !!searchString && m_graph.isWatchingStringSymbolReplaceWatchpoint(m_node)) {
```

Comment:
Regarding this entire code under this branch, we have runtime code that does the same thing. Can we share it with the runtime code? It could also be a FIXME to do it later.

Reply:
Yeah, I'll add FIXME here, and unify them in the next patch.